mlflow/tracing/fluent.py

```diff
@@ -229,10 +240,40 @@ def search_traces(
         max_results: Maximum number of traces desired. If None, all traces matching the search
             expressions will be returned.
         order_by: List of order_by clauses.
+        extract_fields: List of fields to extract from the traces. Fields are of the form
+            "span_name.[inputs|outputs].field_name" or "span_name.[inputs|outputs]". For example,
+            given a trace with a span named "predict" and outputs with names "result" and
+            "explanation", the field "predict.outputs.result" would extract the "result" field, and
+            the field "predict.outputs" would extract the entire outputs dictionary with keys
```

Comment:
@jessechancy This API returns pandas dataframe, does this docstring mean the multiple outputs will be returned as single dictionary column? It might not be super intuitive how the result is converted to dataframe, could you update the docstring to cover following cases at least?

- Default behavior (fields = None)
- Behavior when passing single vs multiple span filed names
- Behavior when no filed name is passed (span_name.inputs)
- Behavior when invalid string is passed (silent return)
- For each, put how the output pandas dataframe looks like in table format.
- Explicitly mention any limitation e.g. only inputs/outputs are supported, any limitation in span name (e.g. can it include period?)